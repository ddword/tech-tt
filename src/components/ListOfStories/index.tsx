import React, {useState} from "react";
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Drawer from '@material-ui/core/Drawer';
import { inherits } from "util";

const drawerWidth = 240;
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    listOfStories: {
        height: "inherit",
    },  
    // necessary for content to be below app bar
    toolbar: theme.mixins.toolbar,
    drawer: {
        width: drawerWidth,
        flexShrink: 0,
    },
    drawerPaper: {
        width: drawerWidth,
    }
  }),
);

type Props = {
    stories: number[],
    selectStory(id:number): void
}

const ListOfStories: React.FC<Props> = ({stories, selectStory}) => {
    stories = stories||[];
    const classes = useStyles();
    const [selectedIndex, setSelectedIndex] = useState<number>();
    
    const onSelect = (story: number, id: number) =>{
       //console.log('Select Story', story)
        selectStory(story);
        setSelectedIndex(id);
    }

    return (
        <div className={classes.listOfStories}>
            <Drawer
            className={classes.drawer}
            variant="permanent"
            classes={{ paper: classes.drawerPaper}}
            anchor="left"
            >
                <div className={classes.toolbar} />
                <Divider />
                <h4 style={{padding:"0 15px"}}>Stories:</h4>
                <List component="nav">
                { stories && stories.map((s,id) =>
                    <ListItem
                        button
                        selected={selectedIndex === id}
                        key={id} 
                        onClick={() => onSelect(s,id) }>
                        <a> { s }</a>   
                    </ListItem>
                )}
                </List>
            </Drawer>
        </div>
    );    
}
export default ListOfStories;
