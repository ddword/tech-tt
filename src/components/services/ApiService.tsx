//Todo write with Provider
const ApiService = {
    // actions from https://github.com/HackerNews/API
    axios: require('axios').default,
    endpoint:'https://hacker-news.firebaseio.com/v0/',

   
    getListOfStories: async function() {
        return await this.axios({
          method: 'get',
          url: `${this.endpoint}topstories.json` + `?print=pretty`,
        })
    },
      
    getStory: async function(id: number) {
        return await this.axios({
            method: 'get',
            url: `${this.endpoint}item/` + id + `.json?print=pretty`,
        })
    }, 
    
    getComment: async function(id:number){
        return await this.axios({
            method: 'get',
            url: `${this.endpoint}item/` + id + `.json?print=pretty`,
        })
    }
}
export default ApiService;