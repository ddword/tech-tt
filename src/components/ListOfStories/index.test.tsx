import React from "react";
import { shallow, mount } from "enzyme";
import ListOfStories from './';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ApiService from '../services/ApiService';

const mockStories = [1,3];

describe('tests in ListOfStories',() => {
    test('renders test react import of childs', () => {
        const wrapper = shallow(<ListOfStories stories={mockStories} selectStory = {Boolean}/>);
        expect(wrapper).toMatchSnapshot();
        expect(ApiService).toBeDefined();
    });
    test('children listItem with story', () => {
        const wrapper = shallow(<ListOfStories stories={mockStories} selectStory = {Boolean}/>)
        expect(wrapper.find('List').children(ListItem).length).toBeLessThanOrEqual(20);
    });
    test('state story', () => {
        const wrapper = shallow(<ListOfStories stories={mockStories} selectStory = {Boolean}/>);
        //expect(wrapper.contains('story')).toBeCalled();
    });
    
});    