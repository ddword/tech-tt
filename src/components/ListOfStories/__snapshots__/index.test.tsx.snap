// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in ListOfStories renders test react import of childs 1`] = `ShallowWrapper {}`;
